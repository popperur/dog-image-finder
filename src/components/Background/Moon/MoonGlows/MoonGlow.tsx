import { MoonGlowHolder } from "components/Background/Moon/MoonGlows/styles.tsx";

interface MoonGlowProps {
  top: number;
  left: number;
  width: number;
  height: number;
  animationDuration: number;
  animationDelay: number;
}

function MoonGlow({
  top,
  left,
  width,
  height,
  animationDuration,
  animationDelay,
}: MoonGlowProps) {
  const stylesArray = [
    `top: ${top}px`,
    `left: ${left}px`,
    `width: ${width}px`,
    `height: ${height}px`,
    `animation-duration: ${animationDuration}s`,
    `animation-delay: ${animationDelay}s`,
  ];

  return <MoonGlowHolder $styles={stylesArray} />;
}

export default MoonGlow;
