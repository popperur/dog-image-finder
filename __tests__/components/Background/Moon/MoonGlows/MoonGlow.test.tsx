import "jest-styled-components";
import { render, RenderResult } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import MoonGlow from "components/Background/Moon/MoonGlows/MoonGlow";

describe("MoonGlow component", () => {
  let rendered: RenderResult;

  beforeEach(() => {
    rendered = render(
      <MoonGlow
        top={10}
        left={20}
        width={40}
        height={60}
        animationDuration={2}
        animationDelay={4}
      />,
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds the styles to the styled component", () => {
    const { container } = rendered;

    const moonGlowHolder = container.firstChild;

    expect(moonGlowHolder).toHaveStyleRule("top", "10px");
    expect(moonGlowHolder).toHaveStyleRule("left", "20px");
    expect(moonGlowHolder).toHaveStyleRule("width", "40px");
    expect(moonGlowHolder).toHaveStyleRule("height", "60px");
    expect(moonGlowHolder).toHaveStyleRule("animation-duration", "2s");
    expect(moonGlowHolder).toHaveStyleRule("animation-delay", "4s");
  });
});
