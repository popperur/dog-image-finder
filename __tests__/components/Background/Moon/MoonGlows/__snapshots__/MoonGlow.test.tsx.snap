// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoonGlow component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: xvTUm infinite;
  top: 10px;
  left: 20px;
  width: 40px;
  height: 60px;
  animation-duration: 2s;
  animation-delay: 4s;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
